sympy/physics/mechanics/tests/test_joint.py

```diff
@@ -0,0 +1,239 @@
+from sympy import Symbol
+from sympy.physics.vector import ReferenceFrame, Point
+from sympy.physics.mechanics import inertia, RigidBody, KanesMethod, \
+    dynamicsymbols
+
+from pydy.bodies import Body
+from pydy.joints import PinJoint, SlidingJoint, CylindricalJoint
+from pydy.joints_method import JointsMethod
+
+
+class TestPinJoint():
+    def setup(self):
+        parent = Body('parent')
+        child = Body('child')
+
+        gravity = Symbol('gravity')
+        # add gravity force in y direction at the center of mass of child.
+        child.add_force((0, child.get_mass() * gravity, 0), child.get_masscenter())
+
+        self.pin_joint = PinJoint('pin_joint', parent, child, parent_point_pos=(1, 1, 1),
+                             child_point_pos=(1, 1, 1))
+
+        JM = JointsMethod([self.pin_joint], parent)
+        self.eom = JM.rhs()
```

Comment:
Shouldn't tests for `JointsMethod` be with that class and PR? I'd rather see the joint classes function independently of JointsMethod and have only it tested in this PR.